library/alloc/src/lib.rs

```diff
@@ -72,7 +72,7 @@
 // Lints:
 #![deny(unsafe_op_in_unsafe_fn)]
 #![deny(fuzzy_provenance_casts)]
-#![warn(deprecated_in_future)]
+#![warn(deprecated)]
```

Comment:
unsure: seems like we wouldn't want this?  Is it just temporary while figuring out other things?